examples/singletons.rb

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env ruby
+example_description = <<DESC
+
+  Sub Plans Example
+  ===================
+
+  This example shows, how singleton actions can be used for making sure
+  there is only one instance of the action running at a time.
+
+  Singleton actions try to obtain a lock at the beggining of their plan
+  phase and fail if they can't do so. In run phase they check if they
+  have the lock and try to acquire it again if they don't. These actions
+  release the lock at the end of their finalize phase.
+
+DESC
+
+require_relative 'example_helper'
+
+class SingletonExample < Dynflow::Action
+  include Dynflow::Action::Singleton
+
+  def run
+    with_valid_singleton_lock do
```

Comment:
What about encapsulating the locking calls into a middleware, so that the actions developer don't have to care/think not to forget to put the right lock/unlock calls at appropriate places?